config-model/src/main/java/com/yahoo/schema/RankProfile.java

```diff
@@ -1265,14 +1265,45 @@ private Optional<TensorType> resolveOnnxInputType(String onnxInputName, OnnxMode
         return Optional.empty();  // if this context does not contain this input
     }
 
+    private static class AttributeErrorType extends TensorType {
+        private final DeployLogger deployLogger;
+        private final String attr;
+        private final Attribute.CollectionType collType;
+        private boolean shouldWarn = true;
+        AttributeErrorType(DeployLogger deployLogger, String attr, Attribute.CollectionType collType) {
+            super(TensorType.Value.DOUBLE, List.of());
+            this.deployLogger = deployLogger;
+            this.attr = attr;
+            this.collType = collType;
+        }
+        private void warnOnce() {
+            if (shouldWarn) {
+                deployLogger.log(Level.WARNING, "Using attribute(" + attr +") " + collType + " as ranking expression input");
```

Comment:
Perhaps the message should say something about what this implies, so that the user can do some changes.